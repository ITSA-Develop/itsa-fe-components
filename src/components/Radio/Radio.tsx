// TODO: waiting for final designs

import type { RadioGroupProps } from 'antd';
import { Radio as AntRadio } from 'antd';

export interface IRadioProps extends RadioGroupProps {
	label?: string;
	variant?: 'default' | undefined;
}
export const Radio = ({ label, ...rest }: IRadioProps) => {
	const { variant, className, rootClassName, ...props } = rest;
	const mergedRootClassName = [
		rootClassName,
		variant === 'default' ? 'itsa-radio--default' : undefined,
	]
		.filter(Boolean)
		.join(' ');

	return (
		<div>
			{label && <p>{label}</p>}
			<AntRadio.Group
				{...props}
				className={className}
				rootClassName={mergedRootClassName}
			/>
		</div>
	);
};
