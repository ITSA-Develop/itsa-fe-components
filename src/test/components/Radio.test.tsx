import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Radio } from '../../components/Radio/Radio';

describe('Radio component', () => {
	it('renders label when provided', () => {
		const { container } = render(<Radio label="Choose option" options={[{ label: 'Option 1', value: 1 }]} />);
		expect(screen.getByText('Choose option')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders radio options and allows selection', async () => {
		const user = userEvent.setup();
		const { container } = render(
			<Radio
				options={[
					{ label: 'Option 1', value: 1 },
					{ label: 'Option 2', value: 2 },
				]}
				defaultValue={1}
			/>,
		);

		const option2 = screen.getByLabelText('Option 2');
		expect(option2).toBeInTheDocument();
		await user.click(option2);
		expect(option2).toBeChecked();
		expect(container).toMatchSnapshot();
	});
});
