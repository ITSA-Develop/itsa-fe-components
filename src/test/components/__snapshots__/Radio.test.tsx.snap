// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Radio component > renders label when provided 1`] = `
<div>
  <div>
    <p>
      Choose option
    </p>
    <div
      class="ant-radio-group ant-radio-group-outline css-dev-only-do-not-override-l9pxc0"
    >
      <label
        class="ant-radio-wrapper css-dev-only-do-not-override-l9pxc0"
      >
        <span
          class="ant-radio ant-wave-target"
        >
          <input
            class="ant-radio-input"
            name="test-id"
            type="radio"
            value="1"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span
          class="ant-radio-label"
        >
          Option 1
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`Radio component > renders radio options and allows selection 1`] = `
<div>
  <div>
    <div
      class="ant-radio-group ant-radio-group-outline css-dev-only-do-not-override-l9pxc0"
    >
      <label
        class="ant-radio-wrapper css-dev-only-do-not-override-l9pxc0"
      >
        <span
          class="ant-radio ant-wave-target"
        >
          <input
            checked=""
            class="ant-radio-input"
            name="test-id"
            type="radio"
            value="1"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span
          class="ant-radio-label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ant-radio-wrapper ant-radio-wrapper-checked css-dev-only-do-not-override-l9pxc0"
      >
        <span
          class="ant-radio ant-wave-target ant-radio-checked"
        >
          <input
            class="ant-radio-input"
            name="test-id"
            type="radio"
            value="2"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span
          class="ant-radio-label"
        >
          Option 2
        </span>
      </label>
    </div>
  </div>
</div>
`;
